pkg/kubectl/cmd/cp/cp.go

```diff
@@ -118,28 +129,43 @@ var (
 )
 
 func extractFileSpec(arg string) (fileSpec, error) {
-	if i := strings.Index(arg, ":"); i == -1 {
+	if len(arg) == 0 {
+		return fileSpec{}, errFileCannotBeEmpty
+	}
+	switch arg[0] {
+	case '.', '/', ':':
```

Comment:
I'm a bit worried that this might cause problems depending on the OS being used. If you want to introduce something like this it would have to be system dependant.

Reply:
Hmm. Not sure what you refer to here. If you see the `TestExtractFileSpec`, you can look at the tests cases.

The reason why the math these three chars is that they isn't valid in a podspec `[[namespace/]pod:]`, so then we know it has to be a file. Is this assumption is wrong?

I guess another alternative to this would be to use a regex to match if the value before the first colon is a valid podspec, `[[namespace/]pod]`.

I don't know that much about windows, but adding a backslash here would fix most of that. Handling the `C:` and `D:` part would be more difficult tho, as the matching will have some ambiguity, so someone with more windows experience should take a look at that.

Reply:
What do you think @soltysh?